src/main/java/Duke.java

```diff
@@ -1,10 +1,205 @@
+     */
+    public static void printList() {
+        for (Task task : tasks) {
+            printTask(task);
+        }
+        Greeting.printHorizontalLines(isSinglish);
+    }
+
+    /**
+     * Marks or Unmark the selected task whether it is done, prints out the selected task alongside its state.
+     * If the user gives an invalid index, informs the user about it.
+     * Does nothing if the user trys to mark a marked task and vice versa.
+     *
+     * @param index The index of the task selected to be marked or unmarked.
+     * @param isMark Whether to mark or unmark the task.
+     */
+    public static void markTask(int index, boolean isMark) {
+        index--;
+        if (index < 0 || index >= tasks.size()) {
+            Greeting.warnOutOfRange(isSinglish);
+        } else {
+            if (tasks.get(index).getIsDone() != isMark) {
+                tasks.get(index).switchIsDone();
+            }
+            Greeting.sayUpdatedTask(isSinglish);
+            tasks.get(index).printTask();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Greeting.sayHello(isSinglish);
+
+        while (true) {
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+
+            String[] commands = line.split(" ");
+            boolean isBye = commands[0].equals(BYE_COMMAND);
+            boolean isValidChangeLangCommand = commands[0].equals(CHANGE_COMMAND) && commands.length == 2 && commands[1].equals(LANG_COMMAND);
+            boolean isValidPrintListCommand = commands[0].equals(LIST_COMMAND) && commands.length == 1;
+            boolean isValidMarkOrUnmarkCommand = (commands[0].equals(MARK_COMMAND) || commands[0].equals(UNMARK_COMMAND)) && commands.length == 2;
+            boolean isTodo = commands[0].equals(TODO_COMMAND);
+            boolean isDeadline = commands[0].equals(DEADLINE_COMMAND);
+            boolean isEvent = commands[0].equals(EVENT_COMMAND);
+
+            if (isBye) {
+                Greeting.sayGoodbye(isSinglish);
+                break;
+            } else if (isValidChangeLangCommand) {
+                changeLanguage();
+            } else if (isValidPrintListCommand) {
+                printList();
+            } else if (isValidMarkOrUnmarkCommand) {
+                checkAndMarkTask(commands);
+            } else if (isTodo) {
+                addTodo(line);
+            } else if (isDeadline) {
+                addDeadline(line);
+            } else if (isEvent) {
+                addEvent(line);
+            } else {
+                Greeting.warnWrongSyntax(isSinglish);
+            }
+        }
+    }
+
```

Comment:
The code written for the handling of user input is quite clean and easy to understand. Good job!